plugins/generic/externalFeed/ExternalFeedPlugin.inc.php

```diff
@@ -1,375 +1,329 @@
-						}
-
-						$journalSettingsDao = DAORegistry::getDAO('JournalSettingsDAO');
-						$journalSettings =& $journalSettingsDao->getSettings($journalId);
-
-						$templateMgr->assign('journalSettings', $journalSettings);
-						$externalFeedForm->display();
-					}
-				} else {
-					$request->redirect(null, 'manager', 'plugin', array('generic', $this->getName(), 'feeds'));
-				}
-				return true;
-			case 'settings':
-				$this->import('ExternalFeedSettingsForm');
-				$form = new ExternalFeedSettingsForm($this, $journal->getId());
-				if ($request->getUserVar('save')) {
-					$request->redirect(null, 'manager', 'plugin', array('generic', $this->getName(), 'feeds'));
-				} elseif ($request->getUserVar('uploadStyleSheet')) {
-					$form->uploadStyleSheet();
-				} elseif ($request->getUserVar('deleteStyleSheet')) {
-					$form->deleteStyleSheet();
-				}
-				$form->initData();
-				$form->display();
-				return true;
-			case 'feeds':
-			default:
-				$this->import('ExternalFeed');
-				$rangeInfo = Handler::getRangeInfo($request, 'feeds');
-				$externalFeedDao = DAORegistry::getDAO('ExternalFeedDAO');
-				$feeds =& $externalFeedDao->getExternalFeedsByJournalId($journalId, $rangeInfo);
-				$templateMgr->assign('feeds', $feeds);
-
-				$templateMgr->display($this->getTemplatePath() . 'externalFeeds.tpl');
-				return true;
-		}
-	}
-}
-
-?>
+<?php
+
+/**
+ * @file plugins/generic/externalFeed/ExternalFeedPlugin.inc.php
+ *
+ * Copyright (c) 2013-2018 Simon Fraser University Library
+ * Copyright (c) 2003-2018 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ExternalFeedPlugin
+ * @ingroup plugins_generic_externalFeed
+ *
+ * @brief ExternalFeedPlugin plugin class
+ */
+
+import('lib.pkp.classes.plugins.GenericPlugin');
+
+class ExternalFeedPlugin extends GenericPlugin {
+
+	const CUSTOM_STYLESHEET = 'externalFeedStyleSheet.css';
+
+	/**
+	 * Called as a plugin is registered to the registry
+	 * @param $category String Name of category plugin was registered to
+	 * @return boolean True iff plugin initialized successfully; if false,
+	 * 	the plugin will not be registered.
+	 */
+	public function register($category, $path) {
+		$success = parent::register($category, $path);
+
+		if (!Config::getVar('general', 'installed') || defined('RUNNING_UPGRADE')) return true;
+		if ($success && $this->getEnabled()) {
+			$this->import('classes.ExternalFeedDAO');
+			
+			HookRegistry::register('PluginRegistry::loadCategory', array($this, 'callbackLoadCategory'));
+			
+			$externalFeedDao = new ExternalFeedDAO($this->getName());
+			$returner =& DAORegistry::registerDAO('ExternalFeedDAO', $externalFeedDao);
+			
+			HookRegistry::register('Templates::Management::Settings::website', array($this, 'callbackShowWebsiteSettingsTabs'));
+			HookRegistry::register('LoadComponentHandler', array($this, 'setupGridHandler'));
+			
+			$templateMgr =& TemplateManager::getManager();

```

Comment:
References to objects are almost never required anymore, just FYI. Best to remove them unless they're required.